ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/healthcheck/HealthCheckAgent.java

```diff
@@ -82,13 +94,46 @@ public String getName() {
     }
 
     @Override
-    public HealthCheckResponseDetail check(
-        final SimpleKsqlClient ksqlClient,
-        final URI serverEndpoint
-    ) {
+    public HealthCheckResponseDetail check(final HealthCheckAgent healthCheckAgent) {
       final RestResponse<KsqlEntityList> response =
-          ksqlClient.makeKsqlRequest(serverEndpoint, ksqlStatement);
+          healthCheckAgent.ksqlClient
+              .makeKsqlRequest(healthCheckAgent.serverEndpoint, ksqlStatement);
       return new HealthCheckResponseDetail(response.isSuccessful());
     }
   }
+
+  private static class KafkaBrokerCheck implements Check {
+    private final String name;
+
+    KafkaBrokerCheck(final String name) {
+      this.name = Objects.requireNonNull(name, "name");
+    }
+
+    @Override
+    public String getName() {
+      return name;
+    }
+
+    @SuppressFBWarnings("RV_RETURN_VALUE_IGNORED_NO_SIDE_EFFECT")
+    @Override
+    public HealthCheckResponseDetail check(final HealthCheckAgent healthCheckAgent) {
+      final String commandTopic = ReservedInternalTopics.commandTopic(healthCheckAgent.ksqlConfig);
+      boolean isHealthy;
+
+      try {
+        healthCheckAgent.serviceContext
+            .getAdminClient()
+            .describeTopics(Collections.singletonList(commandTopic))
+            .values();
+
+        isHealthy = true;
+      } catch (final KsqlTopicAuthorizationException e) {
+        isHealthy = true;
```

Comment:
I marked authorization errors as healthy, but I am unsure if that is a desired result in this situation for healthcheck.